animal_crossing.py

```diff
@@ -100,25 +96,19 @@ def _stallk_check_sell_triggers(cmd, price, expires_in):
 		cmd.reply(msg)
 
 def _stalk_list_sale_prices(cmd):
-	global cached_row_id
+	current_time = datetime.now(timezone.utc)
 	cur = db.execute('''
 	SELECT sell_price.rowid, *
 	FROM sell_price
 	INNER JOIN user ON sell_price.user_id = user.id
-	WHERE sell_price.rowid > ?
-	ORDER BY sell_price.rowid ASC
-	''', (cached_row_id,))
+	WHERE expiration > ?
+	''', (str(current_time),))
 
 	prices = cur.fetchall()
-	current_time = datetime.now(timezone.utc)
 	results = {}
 	for price in prices:
 		user_id = price['user_id']
 
-		if current_time > dateutil.parser.parse(price['expiration']):
-			cached_row_id = price['rowid']
-			continue
-
 		if user_id in results:
```

Comment:
just realized... shouldn't we be upserting prices so there's only one unexpired per user?
(feel free to ignore for this PR)

Reply:
sorry for another dumb question buuuuuuut  https://www.sqlite.org/lang_altertable.html looks like `ALTER TABLE` can't add a new constraint and it will be not fun to try and actually do anything that resembles a proper "migration." Should we just cut our losses and have everyone re-type their timezone? Can we just dump/import a csv?

Reply:
`CREATE INDEX`, I think? https://www.sqlite.org/lang_createindex.html

Reply:
oh, right, duh. God i hate database shit.